features/notifications/utils/handleForegroundNotification.ts

```diff
@@ -4,12 +4,30 @@ import * as Notifications from "expo-notifications";
 import { Platform } from "react-native";
 import logger from "@utils/logger";
 
+// note(lustig): these two handlers are the same - do we need both?
 const handleiOSForegroundNotification = async (
   notification: Notifications.Notification
 ) => {
+  logger.debug("[Notifications] Handling iOS foreground notification");
   resetNotifications();
-  const account = notification.request.content.data?.["account"];
-  if (account && account !== currentAccount()) {
+  // note(lustig): this is not the senderAddress, but the account to which
+  // the notification is to be received.
+  const recipientAccount = notification.request.content.data?.["account"];
+
+  const senderAccount =
+    /* TODO notification.request.content.data?.["sender"]; */ "todo";
```

Comment:
Todo!

Reply:
I've left this half baked code in intentionally and was planning on a follow up, so wanted to leave breadcrumbs for the logic we need.

You cool leaving or want me to remove and make an Issue instead?